### ISSUE TITLE
Problems creating ec2 instances for:  ~/diveintoansible/Using Ansible with Cloud Services and Containers/AWS with Ansible/02

### ISSUE THREAD

**Issue by whirledpeasLLC** [spurin/diveintoansible#16]
Hi James,

I am going through your class on O'Reilly and liking it. I did run into some problems with the aforementioned chapter regarding creation of EC2 instances. There were a number of them. I was able to get it all working but felt I should share what I did to help everyone.

1. Default VPC
This isn't really a thing anymore since classic EC2 is going away. If a student has deleted the default, as I had, they might not be able to recreate one. That was my case. I had to manually create a vpc then manually create a security group referencing that VPC then I had to provide both the and the **vpc_subnet_id** and the **group_id** using the values I got from the dashboard. also, the ami referenced in the video is not current but that might not matter:
`ec2:
           key_name: ansible
           **group_id: <group id>**
           instance_type: t2.micro
           **image: ami-0b0af3577fe5e3532**
           **vpc_subnet_id: <subnet id>**
           region: us-east-1
           wait: true
           exact_count: 5
           **assign_public_ip: yes**
           count_tag:
              Name: AnsibleNginxWebservers
           instance_tags:
              Name: Ansible
        register: ec2`

2. Error when running the script when building the host group. 
once I added the properties as noted above, the instances would get created but building the group threw this error
`ERROR! Invalid empty host name provided: `
to fix this, I added the **assign_public_ip: yes** property which fixed the issue.

These were my fixes. Not being an expert, there might have been another route to get to success but this worked for me.

references:
[Migrate from EC2-Classic to a VPC](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/vpc-migrate.html)
[VPC Wizard instructions](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-getting-started.html)
[amazon.aws.ec2 – create, terminate, start or stop an instance in ec2](https://docs.ansible.com/ansible/latest/collections/amazon/aws/ec2_module.html)
[source for the public ip property addition](https://www.mindbowser.com/how-to-create-ec2-instances-using-ansible/)

Cheers, Michael.

**Comment by spurin** [comment]
Thanks Michael! ( @whirledpeasLLC )

Appreciate the feedback.  I'm currently working through the 3rd edition of the course.

It's an enhancement mostly to the audio and visuals but, also fixing areas that have aged.

Am just about to start section 6 and others have also raised issues with AWS.  Therefore, most likely I'll give this one a complete re-write.

Thanks for letting me know the above.  I'll share an updated video here as soon as it's available.

Best Regards


James Spurin

**Comment by spurin** [comment]
@whirledpeasLLC currently going through this now, will report on findings.

**Comment by spurin** [comment]
Hi @whirledpeasLLC 

So strangely, I've just gone through this without any issues so am somewhat confused as to why you had problems.  With stating this by the way, I'm not in any way suggesting it didn't happen, but can't really explain or replicate the issues that you encountered.

I also just tried deleting the default vpc and re-creating as per the video -

![image](https://user-images.githubusercontent.com/850464/145017859-e8e59bf5-9b24-45ef-b1b8-5820fea6f78d.png)

Select Create VPC

![image](https://user-images.githubusercontent.com/850464/145017960-58c2657f-be52-4eda-a9e1-6330d5cedc25.png)

Follow Wizard - 

![image](https://user-images.githubusercontent.com/850464/145018001-c9994efe-fcde-4e5e-9717-ed2bf7fd52d6.png)

Recreated -

![image](https://user-images.githubusercontent.com/850464/145018063-96f8f24f-cd9c-429d-ab50-d66eefb04846.png)

Looking at the console I'm using the new EC2 dashboard.

In doing this example, I didn't change anything from the repo so therefore, I kept the existing inventory and I used the ami image that was already referenced.

I know from my own studies that AWS will periodically test new interfaces etc so this may be an issue with bad timing.

Will leave the issue open for now and will follow up to close it later.  Hopefully if you try again, all will be as expected 😃
